packages/charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -0,0 +1,186 @@
+import { select as d3Select } from 'd3-selection';
+import { classNamesFunction, customizable, IClassNames } from '../../Utilities';
+import {
+  IHorizontalBarChartProps,
+  IHorizontalBarChartStyleProps,
+  IHorizontalBarChartStyles,
+  IDataPoint
+} from './HorizontalBarChart.types';
+
+const getClassNames = classNamesFunction<IHorizontalBarChartStyleProps, IHorizontalBarChartStyles>();
+type numericAxis = D3Axis<number | { valueOf(): number }>;
+type stringAxis = D3Axis<string>;
+
+@customizable('HorizontalBarChart', ['theme', 'styles'])
+export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartProps, {}> {
+  private _points: IDataPoint[];
+  private _width: number;
+  private _height: number;
+  private _barHeight: number;
+  private _yAxisTickCount: number;
+  private _colors: string[];
+  private _classNames: IClassNames<IHorizontalBarChartStyles>;
+
+  constructor(props: IHorizontalBarChartProps) {
+    super(props);
+
+    this._points = this.props.data || [];
+
+    this._width = this.props.width || 600;
+    this._height = this.props.height || 350;
+    this._barHeight = this.props.barHeight || 15;
+    this._yAxisTickCount = this.props.yAxisTickCount || 10;
+
+    const { theme } = this.props;
+    const { palette } = theme!;
+    this._colors = this.props.colors || [palette.blueLight, palette.blue, palette.blueMid, palette.blueDark];
+  }
+
+  public render(): JSX.Element {
+    const isNumeric = typeof this._points[0].x === 'number';
+
+    const xAxis = this._createNumericXAxis(isNumeric);
+    const yAxis = isNumeric ? this._createYAxis() : this._createStringYAxis();
+    const bars = isNumeric ? this._createNumericBars() : this._createStringBars();
+
+    const { theme, className, styles } = this.props;
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      width: this._width,
+      height: this._height,
+      className
+    });
+
+    return (
+      <div className={this._classNames.root}>
+        {this.props.chartLabel && <p className={this._classNames.chartLabel}>{this.props.chartLabel}</p>}
+        <svg className={this._classNames.chart}>
+          <g ref={(node: SVGGElement | null) => this._setXAxis(node, xAxis)} className={this._classNames.xAxis} />
+          <g ref={(node: SVGGElement | null) => this._setYAxis(node, yAxis)} className={this._classNames.yAxis} />
+          <g className={this._classNames.bars}>{bars}</g>
+        </svg>
+      </div>
+    );
+  }
+
+  private _setXAxis(node: SVGGElement | null, xAxis: numericAxis | stringAxis): void {
+    if (node === null) {
+      return;
+    }
+    const axisNode = d3Select(node).call(xAxis);
+    axisNode.selectAll('.domain').attr('class', this._classNames.xAxisDomain!);
+    axisNode.selectAll('line').attr('class', this._classNames.xAxisTicks!);
+    axisNode.selectAll('text').attr('class', this._classNames.xAxisText!);
```

Comment:
You shouldn't need exclamation points on any of these classNames usages. (Tied to comment below on interface definition)